app/components/views/TicketsPage/MyTicketsTab/MyTicketsTab.jsx

```diff
@@ -112,6 +131,59 @@ const MyTickets = ({ toggleIsLegacy }) => {
     setSortOrderKey(type.value);
   };
 
+  const arraysEqual = (a, b) => {
+    if (a === b) return true;
+    if (a == null || b == null) return false;
+    if (a.length !== b.length) return false;
+    for (let i = 0; i < a.length; ++i) {
+      if (a[i] !== b[i]) return false;
+    }
+    return true;
+  };
+
+  const handlePageClick = (data) => {
+    setQRPage(data.selected);
+  };
+
+  // getQR asychronously updates the qr codes if current viewed ticket hashes
+  // have changed since last viewing.
+  const getQR = () => {
+    loadingQR = defaultLoading;
+    const filteredTiks = tickets.filter(activeFilter);
+    const newQRHashes = [];
+    for (let i = 0, len = filteredTiks.length; i < len; i++) {
+      newQRHashes.push(filteredTiks[i].txHash);
+    }
+    // Check and stop if ticket hashes have not changed.
+    if (arraysEqual(qrHashes, newQRHashes)) {
+      loadingQR = null;
+      return;
+    }
+    qrHashes.length = 0;
+    qrs.length = 0;
+    setQRPage(0);
+    if (newQRHashes.length == 0) {
+      loadingQR = null;
+      return;
+    }
+    qrs.length = Math.ceil(newQRHashes.length / txPerQR);
+    qrHashes.push.apply(qrHashes, newQRHashes);
+    for (let i = 0; i < qrs.length; i++) {
+      const start = i * txPerQR;
+      const qrdata = JSON.stringify(newQRHashes.slice(start, start + txPerQR));
+      QRCode.toDataURL(qrdata, (err, url) => {
```

Comment:
See comment above about using promise here instead of `(err, url) => {}` callback function.